src/main/java/net/dv8tion/jda/entities/impl/TextChannelImpl.java

```diff
@@ -355,6 +355,38 @@ public TextChannelImpl setPosition(int position)
     {
         return rolePermissionOverrides;
     }
+    
+    @Override
+    public void deleteMessages(List<Message> messages)
+    {
+        if(messages.size() > 100)
+        {
+            throw new IllegalArgumentException("Can't delete more than 100 messages at a time, got " + messages.size());
+        }
+        else if(messages.size() == 1)
+        {
+            messages.get(0).deleteMessage();
```

Comment:
Check if the Bot has the permission required to delete the message (aka if it's not its own msg)?